### ISSUE TITLE
False positive for rule `no-duplicate-selectors` since version `16.8.0`

### ISSUE THREAD

**Issue by xavierfoucrier** [stylelint/stylelint#7892]
### What minimal example or steps are needed to reproduce the bug?

```scss
%link {
  position: relative;
  display: inline-flex;
  font-weight: 500;

  &::before { // ⬅️ line 62
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 1px;
    background: currentcolor;
    opacity: 0.4;
    transition: opacity 400ms ease;
    content: '';
  }

  &:hover {
    &::before {
      opacity: 1;
    }
  }
}
```

### What minimal configuration is needed to reproduce the bug?

```js
export default {
  extends: [
    'stylelint-config-standard-scss',
  ],
};
```

### How did you run Stylelint?

I am running `stylelint` using command line with `stylelint src/**/*.scss`.

### Which Stylelint-related dependencies are you using?

```json
{
    "stylelint": "^16.8.0",
    "stylelint-config-standard-scss": "^13.1.0",
}
```

### What did you expect to happen?

Linter to pass.
> It was the case with version `16.7.0`.

### What actually happened?

The following problems were reported:

```
74:3  ✖  Unexpected duplicate selector "&:hover", first used at line 62    no-duplicate-selectors
75:5  ✖  Unexpected duplicate selector "&::before", first used at line 62  no-duplicate-selectors
```

### Do you have a proposal to fix the bug?

I unfortunately did not dive into the code right now, sorry!

**Comment by xavierfoucrier** [comment]
Fixed by https://github.com/stylelint/stylelint/issues/7890 🚀
